src/env.cpp

```diff
@@ -610,6 +610,9 @@ static void react_to_variable_change(const wcstring &key) {
         invalidate_termsize(true);  // force fish to update its idea of the terminal size plus vars
     } else if (key == L"FISH_READ_BYTE_LIMIT") {
         env_set_read_limit();
+    } else if (key == L"FISH_HISTFILE") {
+        history_destroy();
+        reader_push(history_session_id().c_str());
```

Comment:
I'm not sure. I mean, yeah it looks like it, but where should that pop be? To my knowledge, there's no way to "pop" the history session -- you just set it to a new value which may or not be the previous session.

We can pop the previous reader before pushing the new one here, but I don't know enough about the reader stack to know whether that's the right thing to do.